server.py

```diff
@@ -0,0 +1,169 @@
+import os
+import logging
+import base64
+import cv2
+from imantics import Mask, Polygons
+import numpy as np
+from PIL import Image
+from pathlib import Path
+import json
+from pycococreatortools import pycococreatortools
+from sklearn.model_selection import train_test_split
+import time
+
+from detectron2 import model_zoo
+from detectron2.config import get_cfg
+from detectron2.data.datasets import register_coco_instances
+from detectron2.data import DatasetCatalog, MetadataCatalog, build_detection_test_loader
+from detectron2.engine import DefaultTrainer
+from detectron2.evaluation import COCOEvaluator, inference_on_dataset
+
+from droidlet import dashboard
+if __name__ == "__main__":
+    # this line has to go before any imports that contain @sio.on functions
+    # or else, those @sio.on calls become no-ops
+    dashboard.start()
+from agents.argument_parser import ArgumentParser
+from droidlet.event import sio
+from droidlet.perception.robot import LabelPropagate
+import agents.locobot.label_prop as LP
+log_formatter = logging.Formatter(
+    "%(asctime)s [%(filename)s:%(lineno)s - %(funcName)s() %(levelname)s]: %(message)s"
+)
+
+class OfflineInstance(): 
+
+    def __init__(self): 
+        self.init_event_handlers()
+    
+    def init_event_handlers(self): 
+        @sio.on("offline_label_propagation")
+        def offline_label_propagation(sid, postData): 
```

Comment:
how much of this code is a duplication of the online label prop? can we reduce code duplication here?

Reply:
The differences are in how we read the input rgb/seg data, which account for about a third of the lines in the offline_label_propagation and offline_save_rgb_seg functions. I'll replace the duplicate code and refactor it to the label_prop file.